### ISSUE TITLE
Region enumeration issue

### ISSUE THREAD

**Issue by aleksejlopasov** [vmware/terraform-provider-vra#157]
We use CAS + VMC and faced an issue when using Terraform 
```
  data "vra_data_collector" "this" {
  count = var.data_collector_name != "" ? 1 : 0
  name  = var.data_collector_name
  }

  data "vra_region_enumeration" "this" {
  hostname = var.vcenter_hostname
  password = var.vcenter_password
  username = var.vcenter_username
  dcid     = var.data_collector_name != "" ? data.vra_data_collector.this[0].id : ""
  }

macbook:terraform olopa$ terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.vra_data_collector.this[0]: Refreshing state...
data.vra_region_enumeration.this: Refreshing state...

Error: Post https://api.mgmt.cloud.vmware.com/iaas/api/cloud-accounts-vsphere/region-enumeration: context deadline exceeded

  on cloud-account-vmc.tf line 6, in data "vra_region_enumeration" "this":
   6: data "vra_region_enumeration" "this" {
```
1. Is that ok that **cloud-accounts-vsphere** link is used for VMC? I know that **cloud-accounts-vmc** link also exists and these API endpoints have different POST requests bodies.
2. Even if **cloud-accounts-vsphere** is used, the resource does not cover all the required fields for this endpoint (I'm talking about 'regionIds' field) Why?
3. Using the API client to send the direct request to CAS API returns the following JSON:
```
{
  "message": "Failed to validate credentials. AdapterReference: http://private-cloud-gateway:8000/provisioning/vsphere/endpoint-config-adapter. Error: The Cloud proxy ID is required.",
  "statusCode": 400,
  "errorCode": 0,
  "serverErrorId": "xxxxxxxxxxxxxxxx-xxxx-xxx-xxxxxxxxxx",
  "documentKind": "com:vmware:xenon:common:ServiceErrorResponse"
}
```
What Cloud proxy ID do they mean and why?

**Comment by panguet** [comment]
Hello, If I understood well the context ; you are using a ovf template to install a proxy of VCA in a vcenter  (ref: https://docs.vmware.com/en/VMware-Cloud-Assembly/services/Using-and-Managing/GUID-3F363A84-4F4F-4B77-B38F-547C85F84B7C.html)

Before using Terraform. Are you testing the sync actions (button) or deployment of a Blueprint from with a cloid zone of your vcenter account in VCA ?  (Just to confirm this point and talk )

**Comment by sammcgeown** [comment]
@aleksejlopasov What value are you using for "var.data_collector_name"? It should the the name of the Cloud Proxy you've already deployed in your vSphere environment.

**Comment by aleksejlopasov** [comment]
@sammcgeown correct! I use the cloud proxy name displayed in CAS ui.
@panguet yes, the cloud proxy has been successfully installed using the template. All data is synchronized.

Just to clarify, I don’t have any troubles with data_collector data source. Only region_enumeration throws unexpected error, and as I have already said, I didn’t succeed to get the result even when calling the API endpoint directly with the Insomnia API client.

**Comment by cars** [comment]
It looks like the maybe the most current release of the vra-sdk-go project upstream doesn't have any reference to vmc accounts, while the current swagger spec from https://www.mgmt.cloud.vmware.com/iaas/api/swagger/ does. 

In the client code for [vra-sdk-go](https://github.com/vmware/vra-sdk-go/tree/master/pkg/client/cloud_account)  I see create code for aws,azure, gcp, nsx-t, nsx-v and vsphere cloud accounts, but not for VMC as I'd expect if the current swagger was used to generate client code.  This might entail a new release of vra-sdk-go to include the updates iaas swagger and some additional work on this project to get resolved.

**Comment by dmettem** [comment]
As @cars pointed, a separate data source for vmc region enumeration may solve the issue here.  We will first update the vra-sdk-go with latest swagger specs and prioritize this issue. Thanks for @aleksejlopasov for reporting.

**Comment by dmettem** [comment]
This is fixed in [v0.1.9](https://github.com/vmware/terraform-provider-vra/releases/tag/v0.1.9).
